packages/insomnia/src/ui/routes/project.tsx

```diff
@@ -611,23 +614,39 @@ const ProjectRoute: FC = () => {
     }
   }, [organizationId, permissionsFetcher]);
 
+  useEffect(() => {
+    const isIdleAndUninitialized = storageRuleFetcher.state === 'idle' && !storageRuleFetcher.data && !isScratchpadOrganizationId(organizationId);
+
+    if (isIdleAndUninitialized || lastLoadedOrganizationId !== organizationId) {
+      storageRuleFetcher.load(`/organization/${organizationId}/storage-rule`);
+      // Update the state tracking the last loaded organization ID
+      setLastLoadedOrganizationId(organizationId);
+    }
+  }, [lastLoadedOrganizationId, organizationId, storageRuleFetcher]);
```

Comment:
```suggestion
  useEffect(() => {
    if (!isScratchpadOrganizationId(organizationId)) {
      const load = storageRuleFetcher.load;
      load(`/organization/${organizationId}/storage-rule`);
    }
  }, [organizationId, storageRuleFetcher.load]);
```
In this way, we don't need lastLoadedOrganizationId anymore

Reply:
@CurryYangxx I test it locally and confirm that it works